src/cs/utils/UtilTypes.cs

```diff
@@ -96,8 +96,8 @@ public struct MoneyData {
 
 	// Resets the spending statistics at the end of each round
 	public void NextTurn(int new_budget, int production, int ImportCost) {
+		Budget = Money + new_budget;
```

Comment:
Won't updating the budget like this not take into account the production and building costs from the previous turn, meaning that it will take an entire turn before those costs are felt. This also means that the player essentially will have infinite money on the last turn no?